src/category_theory/limits/shapes/wide_pullbacks.lean

```diff
@@ -0,0 +1,118 @@
+/-
+Copyright (c) 2020 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import data.fintype.basic
+import category_theory.limits.limits
+import category_theory.limits.shapes.finite_limits
+import category_theory.sparse
+import category_theory.punit
+-- import category_theory.connected
+
+universes v u
+
+open category_theory category_theory.limits
+
+variable (J : Type v)
+
+/-- A wide pullback shape for any type `J` can be written simply as `option J`. -/
+@[derive inhabited]
+def wide_pullback_shape := option J
```

Comment:
Do you think it would work to mark this as `@[irreducible]` at the end of the file? (Perhaps it would need a little more API.)

Reply:
It would need more API, but I'm not sure how helpful it would be... What are the advantages of this?

Reply:
It's okay: I don't think there are any particular advantages here specifically. I was just thinking about reducibility a lot yesterday, and wishing a lot more in the library were irreducible!